esp/esdllib/esdl_xpath_extensions_libxml.cpp

```diff
@@ -806,6 +806,138 @@ static void getTxSummary(xmlXPathParserContextPtr ctxt, int nargs)
     xmlXPathReturnString(ctxt, xmlStrdup((const xmlChar*)output.str()));
 }
 
+static void getMaskValueBehavior(xmlXPathParserContextPtr ctxt, int nargs)
+{
+    IEsdlScriptContext *scriptContext = queryEsdlScriptContext(ctxt);
+    if (!scriptContext)
+    {
+        xmlXPathSetError((ctxt), XPATH_INVALID_CTXT);
+        return;
+    }
+    Owned<IDataMaskingProfileContext> masker(scriptContext->getMasker());
+
+    if (nargs < 1 || nargs > 2)
+    {
+        xmlXPathSetArityError(ctxt);
+        return;
+    }
+
+    StringBuffer valueType, maskStyle;
+    xmlChar* tmp;
+    if (2 <= nargs)
+    {
+        tmp = xmlXPathPopString(ctxt);
+        if (xmlXPathCheckError(ctxt))
+            return;
+        maskStyle.append((const char*)tmp);
+        xmlFree(tmp);
+    }
+
+    tmp = xmlXPathPopString(ctxt);
+    if (xmlXPathCheckError(ctxt))
+        return;
+    valueType.append((const char*)tmp);
+    xmlFree(tmp);
+
+    uint8_t matchMask = 0;
+    if (masker && masker->canMaskValue())
+    {
+        static const uint8_t VALUE_TYPE = 1;
+        static const uint8_t MASK_STYLE = 2;
+        static const uint8_t SELECTED = 4;
+        Owned<IDataMaskingProfileValueType> vt(masker->inspector().getValueType(valueType));
```

Comment:
Do you need ownership of the vt, when you have the masker?  Can you use queryValueType(valueType) instead?